app/src/main/java/org/hackillinois/android/view/LoginActivity.kt

```diff
@@ -70,15 +68,36 @@ class LoginActivity : AppCompatActivity() {
                             Log.e("LoginActivity", "Notifications update timed out!")
                         }
                     }
-                    JWTUtilities.writeJWT(applicationContext, it)
-                    runOnUiThread {
+
+                    if (getOAuthProvider() == "google") {
```

Comment:
This if block could be its own function since it's so long

Reply:
There's a lot of nested blocks in this class in general

Reply:
Yeah I'm hoping to remove the nested callbacks with a new Kotlin feature with Retrofit.